codalab/server/bundle_manager.py

```diff
@@ -332,37 +335,32 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run, user_
 
         # Dispatch bundles
         for bundle, bundle_resources in staged_bundles_to_run:
+            codalab_owned_workers = copy.deepcopy(
```

Comment:
I think it'd be less error prone to not have these deepcopy, but make `deduct_worker_resources` do the deepcopy so that all functions behave immutably.

Reply:
Yes, that's true. Updated.